index.js

```diff
@@ -1,47 +1,148 @@
-module.exports = MediaSourceStream
+module.exports = MediaElementWrapper
 
 var inherits = require('inherits')
-var stream = require('stream')
+var stream = require('readable-stream')
+var toArrayBuffer = require('to-arraybuffer')
 
 var MediaSource = typeof window !== 'undefined' && window.MediaSource
 
-inherits(MediaSourceStream, stream.Writable)
+var DEFAULT_BUFFER_DURATION = 60 // seconds
 
-function MediaSourceStream (elem, opts) {
+function MediaElementWrapper (elem, opts) {
   var self = this
-  if (!(self instanceof MediaSourceStream)) return new MediaSourceStream(elem, opts)
-  stream.Writable.call(self, opts)
+  if (!(self instanceof MediaElementWrapper)) return new MediaElementWrapper(elem, opts)
 
   if (!MediaSource) throw new Error('web browser lacks MediaSource support')
-  if (!opts) opts = {}
 
+  self._opts = opts || {}
   self._elem = elem
   self._mediaSource = new MediaSource()
-  self._sourceBuffer = null
-  self._cb = null
+  self._streams = []
+  self.detailedError = null
 
-  self._type = opts.type || getType(opts.extname)
-  if (!self._type) throw new Error('missing `opts.type` or `opts.extname` options')
+  self._errorHandler = function () {
+    self._elem.removeEventListener('error', self._errorHandler)
+    var streams = self._streams.slice()
+    streams.forEach(function (stream) {
+      stream.destroy(self._elem.error)
+    })
+  }
+  self._elem.addEventListener('error', self._errorHandler)
 
   self._elem.src = window.URL.createObjectURL(self._mediaSource)
+}
+
+/*
+ * `obj` can be a previous value returned by this function
+ * or a string
+ */
+MediaElementWrapper.prototype.getStream = function (obj) {
```

Comment:
I'd rename this `.createWriteStream(obj)` to indicate that it's a writable stream (like fs, leveldb does etc)